docs/root/intro/life_of_a_request.rst

```diff
@@ -0,0 +1,631 @@
+decoding <life_of_a_request_http2_decoding>`.
+
+As with the downstream HTTP/2 codec, the upstream codec is responsible for taking Envoy’s standard
+abstraction of HTTP, i.e. multiple streams multiplexed on a single connection with request/response
+headers/body/trailers, and mapping this to the specifics of HTTP/2.
+
+9. TLS transport socket encryption
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The upstream endpoint connection's TLS transport socket encrypts the bytes from the HTTP/2 codec
+output and writes them to a TCP socket for the upstream connection. As with :ref:`TLS transport
+socket decryption <life_of_a_request_tls_decryption>`, in our example the cluster has a transport
+socket configured that provides TLS transport security. The same interfaces exist for upstream and
+downstream transport socket extensions.
+
+.. image:: /_static/lor-client.svg
+   :width: 70%
+   :align: center
+
+10. Response path and HTTP lifecycle
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The request, consisting of headers, and optional body and trailers, is proxied upstream, and the
+response is proxied downstream. The response passes through the HTTP and network filters in the
+opposite order from the request.
+
+Various callbacks for decoder/encoder request lifecycle events will be invoked in HTTP filters, e.g.
+when response trailers are being forwarded or the request body is streamed. Similarly, read/write
+network filters will also have their respective callbacks invoked as data continues to flow in both
+directions during a request.
+
+:ref:`Outlier detection <arch_overview_outlier_detection>` status for the endpoint is revised as the
+request progresses.
+
+A request completes when the upstream response reaches its end-of-stream, i.e. when trailers or the
+response header/body with end-stream set are received. This is handled in
+``Router::Filter::onUpstreamComplete()``.
+
+It is possible for a request to terminate early. This may be due to (but not limited to):
+
+* Request timeout.
+* Upstream endpoint steam reset.
+* HTTP filter stream reset.
+* Circuit breaking.
+* Unavailability of upstream resources, e.g. missing a cluster for a route.
+* No healthy endpoints.
+* DoS protection.
+* HTTP protocol violations.
+* Local reply from either HCM or an HTTP filter. E.g. a rate limit HTTP filter returning a 429 response.
+
+If any of these occur, Envoy may either send an internally generated response, if upstream response
+headers have not yet been sent, or will reset the stream, if response headers have already been
+forwarded downstream. The Envoy :ref:`debugging FAQ <faq_overview_debug>` has further information on
+interpreting these early stream terminations.
+
+11. Post-request processing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Once a request completes, the stream is destroyed. The following also takes places:
+
+* :ref:`Statistics <arch_overview_statistics>` are updated (e.g. timing, active requests, upgrades,
```

Comment:
There are some stats that have been updated prior to this. So might be worth clarifying?